src/allmain.c

```diff
@@ -931,6 +931,8 @@ moveloop()
 				youmonst.movement -= 3;
 			} else if(uwep && uwep->oartifact == ART_TOBIUME){
 				youmonst.movement -= 4;
+			} else if(uarmf && uarmf->oartifact == ART_SEVEN_LEAGUE_BOOTS){
```

Comment:
This should be earlier in the else-if chain. You shouldn't slow down if wielding Tobiume.